src/gil.rs

```diff
@@ -220,6 +232,11 @@ pub struct GILPool {
 impl GILPool {
     /// # Safety
     /// This function requires that GIL is already acquired.
+    ///
+    /// Additionally, when this GILPool is dropped, all PyO3 owned references created since this
```

Comment:
> all PyO3 owned references created since this

What is problematic is using older `Python` token than this pool and it should be clearly written.